packit/cli/utils.py

```diff
@@ -323,7 +323,9 @@ def get_packit_api(
 
         if package_config.dist_git_base_url and (
             remote_hostname in package_config.dist_git_base_url
+            # vvvvvvvvvvvvvv HOW IS THIS REACHABLE??? vvvvvvvvvvvvvv
             or remote_hostname in DISTGIT_HOSTNAME_CANDIDATES
+            # ^^^^^^^^^^^^^^ HOW IS THIS REACHABLE??? ^^^^^^^^^^^^^^
```

Comment:
If you have a look at the `common_package_config` before this PR, there is **always** a value set, they're taken in this order:

1. override in config
2. packaging tool
3. environment variables (which we use in our deployments)
4. Fedora default…

I cannot see this “fallback” being reached in any other way than in the tests… And I'm not sure it is helpful

Corollary, the way this is wrapped in braces is totally wrong… cause we **need** the `dist_git_base_url` set to reach this anyways, even if its value is not even used /o\

Reply:
I’m actually starting to wonder if it’s not the other line, because of _URL_ vs _hostname_…

Reply:
you are right that the `dist_git_url` is always set, but the `remote_url` doesn't have to match that one and then the other condition is reached..but maybe I misunderstood your question

Reply:
I am not sure I am following your thoughts but looking at my failing local test I would say that we have not enough informations here to solve our problem, the `DISTGIT_HOSTNAME_CANDIDATES` is not sufficient anymore we need something more to be able to build the right instance:

`url` -> 'https://gitlab.com/packit-service/hello-world'
`remote_hostname` -> gitlab.com
`package_config.dist_git_base_url` -> 'https://src.fedoraproject.org/'

since *gitlab.com* was not inside `DISTGIT_HOSTNAME_CANDIDATES`  we used to build an `upstream` local instance with the url 'https://gitlab.com/packit-service/hello-world' now since *gitlab.com* is inside `DISTGIT_HOSTNAME_CANDIDATES` we create a `distgit` instance and this makes fail the job.

Reply:
@lbarcziova I've actually got lost in it myself… Now I know what I meant, it's a contradiction!

1. This part (function in the CLI utils) determines where is the Packit running from:
   * is it upstream?
   * is it downstream?

2. We **always** have dist-git URL / namespace set in the config.

3. The last condition is technically a bug, cause even if the remote doesn't match the dist-git URL (either default from config/environment), this allows to match **any** dist-git.

The condition contradicts itself in a sense of:

> I don't match the dist-git that I should, but I can match any other git-forge holding dist-git namespace…

This whole thing haven't broken till now, because there was only one forge with a dist-git and that was the `src.fedoraproject.org`.

Reply:
I may say something wrong here but to me the second check seems to broaden the range of possibilities.
The `dist_git_base_url` can be just one of the following, suppose it is the `src.fedoraproject.org`, could we be working from one of the others (except gitlab.com that does not match here)? If yes, then the second check makes sense to me, and still it does but we need to tighten together the `dist_git_base_url` with its possible alternatives (in my example gitlab is not an alternative uri).
```
DISTGIT_HOSTNAME_CANDIDATES =      
  "src.stg.fedoraproject.org",
        "gitlab.com",
        "src.fedoraproject.org",
        "pkgs.fedoraproject.org",
        "pkgs.stg.fedoraproject.org",
```